doc/nrf/ug_partition_manager.rst

```diff
@@ -0,0 +1,189 @@
+
+.. _pm_overview:
+
+Overview
+=============
+
+The Partition Manager script reads a set of Partition Manager and Kconfig configurations.
+Using this information it deduces the start address and size of each partition.
+
+In a multi image build context there exist one root image, and one or more sub-images.
+The size of the root image partition is dynamic, while the size of all sub-image partitions are statically defined.
+
+When the start addresses and sizes have been set, the values are used in the preprocessing of the linker script for each image.
+
+You can define three types of partitions:
+ * *Image partitions* are used to set the start address and size in the linker script of an image.
+ * *Placeholder partitions* reserve space like image partitions.
+   However, their start address and size are not used as input to any linker script.
+ * *Phony partitions* are used to get a reference to a flash area.
+   They do not reserve space.
+   A *Sub partition* is an example of a phony partition.
+
+.. _pm_configuration:
+
+Configuration
+=============
+Each sub-image is required to define its Partition Manager configuration in a file called :file:`pm.yml`.
+
+.. note::
+   The root application does not define its own :file:`pm.yml` because its partition size
+   and placement is implied by the size and placement of the sub image
+   partitions.
+
+.. _pm_yaml_format:
+
+Configuration file format
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The format of the :file:`pm.yml` file is as follows:
+
+.. code-block:: yaml
+
+   partition_name:
+      option_dict_name:
+         option_specific_values
+
+
+Where ``option_dict_name`` can be:
+
+placement: dict{}
+   The placement of the partition relative to other partitions, the end of flash, or the root image partition `app`.
+   The placement is formatted as a yaml dict.
+   The valid keywords are listed below.
+
+      before: list
+         Place the partition before the first existing partition in the list.
+
+      after: list
+         Place the partition after the first existing partition in the list.
+
+   The valid values in the lists are `app`, `end`, or the name of any partition.
+   It is not possible to place the partition after `end`.
+
+sub_partitions: list
+   Define sub partitions which span across one or more other partitions.
+   The sub partitions split the total size of the spanned partitions, and therefore have equal size.
+   This property cannot be used together with the `placement` property.
+   When this property is set for a partition, it is required that the `span` property is also set.
+
+span: list
+   When creating sub_partitions, this option lists what partitions the sub_partitions should span across.
+   This property cannot be used together with the `placement` property.
+
+.. _pm_yaml_preprocessing:
+
+Configuration file preprocessing
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Each :file:`pm.yml` file is preprocessed.
```

Comment:
so it possible and useful to put some C processor syntaxt into yaml's

Reply:
Is this a proposal for something I should add to the document? I'm not sure I understand :)

Reply:
yes

Reply:
Now that I added the comments to the example of pre processing I think there is no need to explain the motivation for supporting pre processing here. 

Do you agree?

Reply:
agree - I just express that i like this.